src/lib/routines.sh

```diff
@@ -3,143 +3,95 @@
 function routine() {
   set -euo pipefail
 
-  local _CMD
+  if [[ -n "${XDG_RUNTIME_DIR:-}" ]] && [[ ! -e "${XDG_RUNTIME_DIR:-}" ]]; then
+    # silence warning if $XDG_RUNTIME_DIR does not exist
+    unset XDG_RUNTIME_DIR
+  fi
 
-  _CMD="${1:-}"
+  CAUR_CURRENT_ROUTINE="${1:-}"
 
-  case "${_CMD}" in
-  'hourly')
-    hourly
-    ;;
-  'morning')
-    daily-morning
-    ;;
-  'afternoon')
-    daily-afternoon
-    ;;
-  'nightly')
-    daily-night
-    ;;
-  'midnight')
-    daily-midnight
-    ;;
+  case "${CAUR_CURRENT_ROUTINE}" in
   'tkg-kernels')
     routine-tkg-kernels
     ;;
   'clean-archive')
     clean-archive
     ;;
   *)
-    echo 'Unrecognized routine'
-    return 22
+    generic-routine
```

Comment:
Instead of exporting CAUR_CURRENT_ROUTINE, I prefer having it as local and passed through parameters, otherwise we can't easily reuse this function later...

Reply:
This will not be possible. It is used on the trap handler.